src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/EnvelopeRepository.java

```diff
@@ -14,4 +17,24 @@
      * @return A list of envelopes which belongs to the given jurisdiction.
      */
     List<Envelope> findByJurisdictionAndStatus(String jurisdiction, Status status);
+
+    /**
+     * Finds envelope for a given container, zip file name and status.
+     *
+     * @param container from where container originated
+     * @param zipFileName of envelope
+     * @param status of envelope
+     * @return Optional envelope
+     */
+    @Query("select e from Envelope e"
```

Comment:
You can do exists as you are not doing anything with the fetched envelope

Reply:
something like below

```

 @Query("select CASE WHEN COUNT(e) > 0 THEN true ELSE false END from Envelope e"
         + " where e.container = :container"
         + "   and e.zipFileName = :zip"
         + "   and e.status = :status"
         + " order by e.createdAt desc"
     )
     Optional<Envelope> existsByStatus(
         @Param("container") String container,
         @Param("zip") String zipFileName,
         @Param("status") Status status
     );

```

Reply:
I'm using envelope to log ID and created at. Can skip the log to use exist query or leave as it is now

Reply:
ohh I see `createdAt` might be useful while debugging so should be fine 👍